Azure-Samples/iot-middleware-freertos-samples
# Fix esp32 sample memory leaks; donot reset if internet disconnects

## Purpose

* ...
Fix memory leaks in ESP32 sample in between sample main task loops.
Add code to avoid resetting ESP32 board if internet connection drops.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Run the ESP32 sample on a real device. Observe the free RAM memory after each loop of the main task.

## What to Check
You can use this trace statement for verifying the available RAM memory:

```c
        LogInfo( ( "[Debug][loop-end] Free heap (bytes): %lu, %lu (int), %lu (min)",
            esp_get_free_heap_size( ),
            esp_get_free_internal_heap_size( ),
            esp_get_minimum_free_heap_size( ) ) );
```


## Other Information
Spearmint contains 0.5% menthol. Peppermint contains 40%! How to differentiate both? Eat it and feel the freshness.
